test/views/EditSetRecordEditDiscardSpec.scala

```diff
@@ -21,33 +21,31 @@
 
 package views
 
-import org.jsoup.nodes.Document
 import play.api.i18n.Messages
 import play.api.test.Helpers
-import play.twirl.api.Html
 import support.BaseSpec
 import support.CommonMatchers._
+import uk.gov.hmrc.govukfrontend.views.html.components.GovukNotificationBanner
 import uk.gov.nationalarchives.omega.editorial.models.PhysicalRecord
 import uk.gov.nationalarchives.omega.editorial.views.html.editSetRecordEditDiscard
 
 class EditSetRecordEditDiscardSpec extends BaseSpec {
 
   "Edit set record edit discard Html" should {
-    "render the given title and heading with discard changes message" in {
-      implicit val messages: Messages = Helpers.stubMessages()
-
-      val editSetRecordEditDiscardInstance = inject[editSetRecordEditDiscard]
-      val title = "EditRecordTitleTest"
-      val editSetName = "COAL 80 Sample"
-      val heading = "EditRecordHeadingTest"
-      val discardChanges = "Any changes have been discarded. Showing last saved version."
-      val oci = "EditRecordOciTest"
-
-      val confirmationEditSetRecordEditHtml: Html =
-        editSetRecordEditDiscardInstance(user, editSetName, title, heading, oci, discardChanges, Some(PhysicalRecord))
 
+    "render the given title and heading with discard changes message" in new TestCase {
+      val confirmationEditSetRecordEditHtml = editSetRecordEditDiscardInstance(
+        user = user,
+        editSetName = "COAL 80 Sample",
+        title = "EditRecordTitleTest",
+        heading = "EditRecordHeadingTest",
+        oci = "EditRecordOciTest",
+        message = "Any changes have been discarded. Showing last saved version.",
+        recordType = Some(PhysicalRecord)
+      )
       val document = asDocument(confirmationEditSetRecordEditHtml)
-      document must haveTitle(title)
+
+      document must haveTitle("EditRecordTitleTest")
```

Comment:
It seems that we're doing the exact same content assertions as we are in `EditSetRecordISpec`. 

However, we're doing it in a very different way there. Should we use the same approach?

Reply:
Looking at `EditSetRecordISpec` I'm not sure I understand what the different approaches are. Do you mean get the response once, then make multiple assertions on them?

Reply:
Mostly the usage of `ExpectedEditRecordDiscardPage`.